tests/functional/data/configured-endpoint-urls/test-cases/profile-tests.json

```diff
@@ -0,0 +1,426 @@
+{
```

Comment:
Another thing I noticed... Instead of making a new `data` directory in the functional test suite, let's just make a new directory for `configured-endpoint-urls` and make this JSON file as a direct child in that directory. So the end directory structure would be:
```
tests/functional/configured-endpoint-urls/
    __init__.py
   test_configured_endpoint_url.py
   test-cases.json
```
Mainly I don't think we should be creating a `data` directory which conflates with the model directory naming pattern at the root of the functional test suite just for this set of tests.